### PR TITLE
SubtleCrypto algo

### PR DESCRIPTION
**Pull request by hamishwillee**
This pushes the dictionary content in [RsaOaepParams](https://developer.mozilla.org/en-US/docs/Web/API/RsaOaepParams), [AesCbcParams](https://developer.mozilla.org/en-US/docs/Web/API/AesCbcParams), and [AesCtrParams](https://developer.mozilla.org/en-US/docs/Web/API/AesCtrParams)  into the methods that use them.

I could not delete these. Got error even though stub is correct and file exists. Advice welcome - can I just delete them?
```
$ yarn content delete Web/API/RsaOaepParams
error: document does not exists: /en-US/docs/Web/API/RsaOaepParams
```

EDITED. Everything except deletion done. Ready for review.

### CODE REVIEW

File: files/en-us/web/api/subtlecrypto/encrypt/index.md [hunk 1]
```diff
@@ -33,10 +30,42 @@ encrypt(algorithm, key, data)
...
+
+      - `name`
+        - : A string indicating the name of the algorithm: `AES-CBC`, `AES-GCM`.
+      - `iv`
+        - : An {{jsxref("ArrayBuffer")}}, a {{jsxref("TypedArray")}}, or a {{jsxref("DataView")}}.
+          The initialization vector.
+          Must be 16 bytes, unpredictable, and preferably cryptographically random.
+          However, it need not be secret (for example, it may be transmitted unencrypted along with the ciphertext).
+
+    - To use [AES-CTR](#aes-ctr), pass an object with the properties the following properties: <!-- AesCtrParams dictionary in the spec -->
```
Comment (hamishwillee) [inline files/en-us/web/api/subtlecrypto/encrypt/index.md:55]: So this is the docs from https://developer.mozilla.org/en-US/docs/Web/API/AesCtrParams. Just the properties here which would duplicate to the other docs. 

Here we link down to  `"[AES-CTR](#aes-ctr)"` . In other docs we would link to this doc.

File: files/en-us/web/api/subtlecrypto/encrypt/index.md [hunk 2]
```diff
@@ -95,6 +114,23 @@ The RSA-OAEP public-key encryption system is specified in [RFC 3447](https://dat
 
 This represents AES in Counter Mode, as specified in [NIST SP800-38A](https://csrc.nist.gov/publications/detail/sp/800-38a/final).
 
+AES is a block cipher, meaning that it splits the message into blocks and encrypts it a block at a time.
```
Comment (hamishwillee) [inline files/en-us/web/api/subtlecrypto/encrypt/index.md:117]: This is description from https://developer.mozilla.org/en-US/docs/Web/API/AesCtrParams.

**Comment by wbamberg** [comment]
> This pushes the dictionary content in [RsaOaepParams](https://developer.mozilla.org/en-US/docs/Web/API/RsaOaepParams) and [AesCbcParams](https://developer.mozilla.org/en-US/docs/Web/API/AesCbcParams) into the methods that use them.
> 
>     1. I could not delete these. Got error even though stub is correct and file exists. Advice welcome - can I just delete them?
>        ```
>        $ yarn content delete Web/API/RsaOaepParams
>        error: document does not exists: /en-US/docs/Web/API/RsaOaepParams
>        ```

It's a yari bug: https://github.com/mdn/yari/issues/7196.

I have a horrid workaround.

1) In your mdn/content clone, go to content/node_modules/@mdn/yari/content/document.ts
2) At this bit: https://github.com/mdn/yari/blob/3b27ae779958e160cb02b49e9bb855873d1b2c24/content/document.ts#L607 replace `...roots` with just `roots`

Now it will "work", but will incorrectly duplicate the `/en-US/docs/` bit in the destination in _redirects.text, like this: https://github.com/mdn/content/pull/21143/commits/d16515da5f2473a1128ea9253694bdd61dcea61e . You'll have to fix that manually.

I told you it was horrid. If you have a better one, please lmk :).

**Comment by hamishwillee** [comment]
@wbamberg That is horrible. Proposal - add "To be deleted notes" to those pages and leave them until yari is fixed. Add links from the yari bug for cleanup.

What do you think of my thinking for part "2" above? Worth proceeding with that?

File: files/en-us/web/api/subtlecrypto/unwrapkey/index.md [hunk 3]
```diff
@@ -245,78 +250,95 @@ alongside the password the user supplies.
...
 This must the same IV that was originally used to encrypt
 the key.
 */
-const ivBytes = [212,187,26,247,172,51,37,151,27,177,249,142];
+const ivBytes = [212, 187, 26, 247, 172, 51, 37, 151, 27, 177, 249, 142];
 
 /*
 The wrapped key itself.
 */
-const wrappedKeyBytes = [6,155,182,208,7,141,44,18,3,151,58,126,68,100,252,
```
Comment (hamishwillee) [inline files/en-us/web/api/subtlecrypto/unwrapkey/index.md:260]: This is prettier at work

**Comment by hamishwillee** [comment]
@wbamberg I haven't deleted the docs. Can I ignore that for now - perhaps edit edit the dictionary to mark them as pending deletion and add as a todo to the yari bug?

If you say I have to follow the workaround I can.

Otherwise this is ready for review.

File: files/en-us/web/api/subtlecrypto/unwrapkey/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/subtlecrypto/unwrapkey/index.md:260]: Maybe we should add `js-nolint`, or it will keep coming back.

Comment (hamishwillee) [reply files/en-us/web/api/subtlecrypto/unwrapkey/index.md:260]: I thought prettier would consistently lint - i.e.it won't change again?

File: files/en-us/web/api/subtlecrypto/decrypt/index.md [hunk 4]
```diff
@@ -29,12 +29,45 @@ decrypt(algorithm, key, data)
...
+
+      - `name`
+        - : A string indicating the name of the algorithm: `AES-CBC`, `AES-GCM`.
+      - `iv`
+        - : An {{jsxref("ArrayBuffer")}}, a {{jsxref("TypedArray")}}, or a {{jsxref("DataView")}}.
+          The initialization vector.
+          Must be 16 bytes, unpredictable, and preferably cryptographically random.
+          However, it need not be secret (for example, it may be transmitted unencrypted along with the ciphertext).
+
+    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr), pass an object with the properties the following properties: <!-- AesCtrParams dictionary in the spec -->
```
Comment (wbamberg) [inline files/en-us/web/api/subtlecrypto/decrypt/index.md:57]:
```suggestion
    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr), pass an object with the following properties: <!-- AesCtrParams dictionary in the spec -->
```

(this kind of typo always makes me think of https://www.youtube.com/watch?v=CfW-MPUjC_0 )

File: files/en-us/web/api/subtlecrypto/encrypt/index.md [hunk 5]
```diff
@@ -33,10 +30,41 @@ encrypt(algorithm, key, data)
...
+
+      - `name`
+        - : A string indicating the name of the algorithm: `AES-CBC`, `AES-GCM`.
+      - `iv`
+        - : An {{jsxref("ArrayBuffer")}}, a {{jsxref("TypedArray")}}, or a {{jsxref("DataView")}}.
+          The initialization vector.
+          Must be 16 bytes, unpredictable, and preferably cryptographically random.
+          However, it need not be secret (for example, it may be transmitted unencrypted along with the ciphertext).
+
+    - To use [AES-CTR](#aes-ctr), pass an object with the properties the following properties: <!-- AesCtrParams dictionary in the spec -->
```
Comment (wbamberg) [inline files/en-us/web/api/subtlecrypto/encrypt/index.md:55]:
```suggestion
    - To use [AES-CTR](#aes-ctr), pass an object with the following properties: <!-- AesCtrParams dictionary in the spec -->
```

File: files/en-us/web/api/subtlecrypto/wrapkey/index.md [hunk 6]
```diff
@@ -56,18 +49,47 @@ wrapKey(format, key, wrappingKey, wrapAlgo)
...
+
+      - `name`
+        - : A string indicating the name of the algorithm: `AES-CBC`, `AES-GCM`.
+      - `iv`
+        - : An {{jsxref("ArrayBuffer")}}, a {{jsxref("TypedArray")}}, or a {{jsxref("DataView")}}.
+          The initialization vector.
+          Must be 16 bytes, unpredictable, and preferably cryptographically random.
+          However, it need not be secret (for example, it may be transmitted unencrypted along with the ciphertext).
+
+    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr), pass an object with the properties the following properties: <!-- AesCtrParams dictionary in the spec -->
```
Comment (wbamberg) [inline files/en-us/web/api/subtlecrypto/wrapkey/index.md:76]:
```suggestion
    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr), pass an object with the following properties: <!-- AesCtrParams dictionary in the spec -->
```

File: files/en-us/web/api/subtlecrypto/unwrapkey/index.md (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/subtlecrypto/unwrapkey/index.md:260]: I'm not sure what that means. I would expect Prettier to prettify everything every time it runs. I don't think it remembers what it has seen before.

Comment (wbamberg) [reply files/en-us/web/api/subtlecrypto/unwrapkey/index.md:260]: Oh, no, maybe I misunderstand! You mean Prettier is adding these spaces? Somehow I thought you were complaining that Prettier had removed them and you have to put them back. Forget I said anything.

File: files/en-us/web/api/subtlecrypto/unwrapkey/index.md [hunk 7]
```diff
@@ -51,18 +44,48 @@ unwrapKey(format, wrappedKey, unwrappingKey, unwrapAlgo, unwrappedKeyAlgo, extra
...
-    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr),
-      pass an [`AesCtrParams`](/en-US/docs/Web/API/AesCtrParams) object.
-    - To use [AES-CBC](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-cbc),
-      pass an [`AesCbcParams`](/en-US/docs/Web/API/AesCbcParams) object.
-    - To use [AES-GCM](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-gcm),
-      pass an [`AesGcmParams`](/en-US/docs/Web/API/AesGcmParams) object.
-    - To use [AES-KW](/en-US/docs/Web/API/SubtleCrypto/wrapKey#aes-kw),
-      pass the string `"AES-KW"` or an object of the form `{ "name": "AES-KW }`.
+
+  - : An object specifying the [algorithm](/en-US/docs/Web/API/SubtleCrypto/encrypt#supported_algorithms) to be used to encrypt the exported key, and any extra parameters as required:
```
Comment (wbamberg) [inline files/en-us/web/api/subtlecrypto/unwrapkey/index.md:48]:
```suggestion
  - : An object specifying the [algorithm](#supported_algorithms) to be used to unwrap the exported key, and any extra parameters as required:
```

(not you, but this is surely wrong)

File: files/en-us/web/api/subtlecrypto/wrapkey/index.md [hunk 8]
```diff
@@ -56,18 +49,47 @@ wrapKey(format, key, wrappingKey, wrapAlgo)
...
-    - To use [AES-CTR](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-ctr),
-      pass an [`AesCtrParams`](/en-US/docs/Web/API/AesCtrParams) object.
-    - To use [AES-CBC](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-cbc),
-      pass an [`AesCbcParams`](/en-US/docs/Web/API/AesCbcParams) object.
-    - To use [AES-GCM](/en-US/docs/Web/API/SubtleCrypto/encrypt#aes-gcm),
-      pass an [`AesGcmParams`](/en-US/docs/Web/API/AesGcmParams) object.
-    - To use [AES-KW](#aes-kw),
-      pass the string `"AES-KW"`, or an object of the form `{ "name": "AES-KW }`.
+
+  - : An object specifying the [algorithm](/en-US/docs/Web/API/SubtleCrypto/encrypt#supported_algorithms) to be used to encrypt the exported key, and any required extra parameters:
```
Comment (wbamberg) [inline files/en-us/web/api/subtlecrypto/wrapkey/index.md:53]:
```suggestion
  - : An object specifying the [algorithm](#supported_algorithms) to be used to wrap the exported key, and any required extra parameters:
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @hamishwillee ! I don't think we should leave notes reminding us to delete, I'll do it in a follow-up.

File: files/en-us/web/api/subtlecrypto/decrypt/index.md (hunk 4 above)
Comment (hamishwillee) [reply files/en-us/web/api/subtlecrypto/decrypt/index.md]: Except it looks like I did it more than just two times. Need face-palm emoticon.

**Pull request merged**